compliance/virtualmachines/agent/README.md

```diff
@@ -0,0 +1,65 @@
+# VM Agent
+
+Runs inside VMs to scan for vulnerabilities and report back to the host via vsock.
+While not directly related to the `compliance` feature, the agent utilizes compliance
+node scanning code for package scanning in the virtual machine.
+
+## What it does
+
+Scans the VM for installed packages (RPM/DNF databases), creates vulnerability reports, and sends them to the host over vsock. Can run once or continuously in daemon mode.
+
+## Usage
+
+```bash
+# Single scan
+sudo ./agent
+
+# Daemon mode (scans every 5 minutes)
+sudo ./agent --daemon
+
+# Custom settings
+sudo ./agent --daemon --index-interval 10m --host-path /custom/path --port 2048
+```
+
+## Flags
+
+- `--daemon` - Run continuously (default: false).
+- `--index-interval` - Time between scans in daemon mode (default: 5m).
+- `--host-path` - Where to look for package databases (default: /).
+- `--port` - VSock port (default: 1024).
+- `--repo-cpe-url` - URL for the repository to CPE mapping.
+- `--timeout` - VSock client timeout when sending index reports.
+- `--verbose` - Prints the index reports to stdout.
+
+## How it works
+
+1. Scans filesystem for RPM/DNF package databases.
+2. Pulls repo-to-CPE mappings from Red Hat.
```

Comment:
I would add a hint that access to Internet or Sensor is required for the operation due to that pull.